src/commands/deploy/deploy.js

```diff
@@ -462,10 +463,26 @@ const deploy = async (options, command) => {
   await command.authenticate(options.auth)
 
   let siteId = options.site || site.id
+
   let siteData = {}
   if (siteId) {
     try {
-      siteData = await api.getSite({ siteId })
+      const [{ siteError, siteFoundById }, [siteFoundByName]] = await Promise.all([
+        api
+          .getSite({ siteId })
+          .then((data) => ({ siteFoundById: data }))
+          .catch((error_) => ({ siteError: error_ })),
+        api.listSites({ name: options.site, filter: 'all', per_page: 1 }),
+      ])
+
+      if (siteFoundById) {
+        siteData = siteFoundById
+      } else if (siteFoundByName && siteFoundByName.name === options.site) {
```

Comment:
**Note for Reviewer:**

The additional `siteFoundByName.name === options.site` check is here since the API can return partially matching results and we want exact matches only.

Reply:
This is a great catch!
What happens if the we have multiple matches, the first one being a partial match?
I don't think the order is guaranteed.

So we retrieve all matches using https://github.com/netlify/cli/blob/bfadc7658518577c069d5e92baddf346f92dca63/src/lib/api.js#L85 and check for an exact match?

Reply:
If the order isn't guaranteed, then we should check multiple sites.

Plus, I don't think it should cause a performance issue since we're filtering by site name which should limit the number of results returned significantly.